File: go/003.go
func lengthOfLongestSubstring(s string) int {
    size, exist, ret := len(s), make([]bool, 128), 0
    for i, j := 0, 0; i < size; i++ {
        for exist[s[i]] {
            exist[s[j]] = false
            j++
        }
        exist[s[i]] = true
        ret = max(ret, i-j+1)
    }
    return ret
}

func max(i, j int) int {
    if i < j {
        return j
    }
    return i
}

File: cpp/003.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int size = s.size(), ret = 0;
        int exist[128] = {0};
        for (int i = 0, j = 0; i < size; ++i) {
            while (exist[s[i]]) 
                exist[s[j++]] = 0;
            exist[s[i]] = 1;
            ret = max(ret, i - j + 1);
        }
        return ret;
    }
};
